core/controllers/profile_test.py

```diff
@@ -1168,7 +1168,7 @@ def raise_exception() -> None:
                     'email': 'email@example.com',
                     'tag': 'Web',
                     'name': 'Name'
-                }, csrf_token=csrf_token, expected_status_int=500)
+                }, csrf_token=csrf_token, expected_status_int=405)
```

Comment:
@mon4our I don't understand why this is throwing a 405. Isn't PUT in the list of allowed methods for MailingListSubscriptionHandler? What exactly was causing the original 500 error?

Reply:
PUT is indeed allowed for MailingListSubscriptionHandler but the `'name'` parameter passed in this test is extra, which causes a TypeError to be thrown. Since the `handle_exception()` method in `base.py` checks for a TypeError before throwing a 405 status code, the request for this test also gets a 405.
```
test_email_provider_error (core.controllers.profile_test.MailingListSubscriptionHandlerTests) ...
 ERROR:root:Exception raised at http://localhost/mailinglistsubscriptionhandler:
 raise_exception() got an unexpected keyword argument 'name'
 
TypeError: raise_exception() got an unexpected keyword argument 'name'
ERROR:root:Exception raised: raise_exception() got an unexpected keyword argument 'name'
``` 
For more context:
The if condition before raising a 405 status code checks for a TypeError because when I tried sending requests with unsupported methods (like POST/PUT for an endpoint that only supports GET) via postman, the exception raised was always a TypeError.
I was not able to add a check for supported methods inside the if as it was breaking the code.
P.S. I have marked the [earlier review](https://github.com/oppia/oppia/pull/20544#discussion_r1650156797) as unresolved so other reviewers can also see the context for these changes. I'll also add comments in the code to prevent confusion if you agree with these changes.

Reply:
@mon4our, is the "name" parameter required for this request? Wondering if the issue is the endpoint vs the error.

Reply:
@adhiamboperes This seems weird. I removed the 'name' parameter from the `put_json` in the test
![image](https://github.com/user-attachments/assets/5e5c97e4-37f2-40c5-94ee-e0a1ebd4e244)
And on rerunning the test, I still get an exception for 'name'
```
{"error": "raise_exception() got an unexpected keyword argument 'name'", "status_code": 405}
```

Reply:
I just realised, the unexpected keyword argument 'name' is not for the route `/mailinglistsubscriptionhandler` but for the `raise_exception()` method defined in this test itself. I added `*args, **kwargs` to the params of the method so that an exception is not thrown when `raise_exception()` is called (I'm pretty sure it doesn't matter what params are passed to the args, kwargs as we won't be using them anyway)  and replaced the `expected_status_int` to the original 500 so that the test looks like:
![image](https://github.com/user-attachments/assets/b98ef300-5f66-40be-87de-7bfe79b08340)
On running this, we get no errors and all the backend tests pass successfully.

Reply:
It does seem like the swap function is passing the keyword which  `raise_exception` does not accept.
Isn't it sufficient to just have 
```python
def raise_exception(**kwargs) -> None:
    raise Exception('Backend error')
```
as opposed to (*args, **kwargs: True)?

Reply:
I'll defer to @seanlip for what the correct approach for that should be. Thanks.